offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -259,8 +263,8 @@ class Offer {
         return items.sum {calculateItemOverhead(it)} as double
     }
 
-    private double calculateItemNet(ProductItem item) {
-        double unitPrice = selectedCustomerAffiliation.category == AffiliationCategory.INTERNAL ? item.product.internalUnitPrice : item.product.externalUnitPrice
+    private BigDecimal calculateItemNet(ProductItem item) {
+        BigDecimal unitPrice = selectedCustomerAffiliation.category == AffiliationCategory.INTERNAL ? item.product.internalUnitPrice : item.product.externalUnitPrice
```

Comment:
Do you need the BigDecimal here? The unit price is specified by hand and thus always rounded to 2 digits after the comma. It might suffice to only specify the function output to be BigDecimal

Reply:
It does not suffice since `double` * `BigDecimal` results in a `double`